### ISSUE TITLE
DllImport and current directory

### ISSUE THREAD

**Issue by xlegalles** [dotnet/corefx#28591]
We need to start several processes from a single application (all .NET Core 2.0 console app) using `Process.Start` and `ProcessStartInfo`. First, we found this  [issue](https://github.com/dotnet/corefx/issues/28590) about `ProcessStartInfo.WorkingDirectory` which seems to not work. Now that we use `Directory.SetCurrentDirectory` in the started application, .NET assemblies are well resolved/loaded but we have another error with a `DllImport` which cannot load the native dll:
`System.DllNotFoundException: Unable to load DLL '***' : The specified module could not be found. (Exception from HRESULT: 0x8007007E)`

The dll is of course in the same directory as its executable. I have tried to add a `SetCurrentDirectory` + `SetDllDirectory` in the Main method w/o any effects. And also added the directory in the PATH.

Of course, everything "works on my machine" (windows 10, .NET Core SDK 2.1.103) and it only happens in the staging env (windows server 2016 .NET Core SDK 2.1.101).

So to summarize:

- If the app is started from the CLI using `dotnet ***.dll`, it _works from any directory_

- If the app is started the same way from another process using `Process.Start()`, it **fails**

- If the app is published as an executable, it _works using the CLI from its current directory_
But it **fails from the parent directory** and of course, also using `Process.Start()` (which is the excpected use case)

Last detail: the DllImport is done in the context of an `Activator.CreateInstance` (using Windsor)

**Comment by danmosemsft** [comment]
@jeffschwMSFT can someone on your team help unravel this?

**Comment by xlegalles** [comment]
I tried to make a simple app to reproduce the problem and discovered that the problem was solved if all C++ dependencies (VC140) are also in the executable directories: so it may be more related to the installation of the C++ runtime than anything else. So I'm sorry and confused about that!
Although, because it worked in some situations, it was a bit confusing.
